server/dav/backend/directory.js

```diff
@@ -154,13 +162,24 @@ var jsDAV_Chunked_Directory = module.exports = jsDAV_FS_Directory.extend(jsDAV_F
           handler.dispatchEvent("afterBind", handler.httpRequest.url,
                                 Path.join(self.path, filename));
 
-          Etag(originalPath, cbfswritechunk);
+          self.getETag(cbfswritechunk);
         });
       } else {
         cbfswritechunk(null, null);
       }
     });
 
     handler.getRequestBody(type, stream, false, function() {});
+  },
+
+  // We redefine fsext's `delete` because it uses asyncjs rmtree, which causes
+  // stack overflows on very large directories.
+  "delete": function(cbfsdel) {
+    var self = this;
+    ChildProcess.spawn("rm", ["-rf", this.path]).on("exit", function(err) {
```

Comment:
Also is there any chance to turn the recursive call into something that iterates, and then keep it all pure-node, and upstream a fix?

e.g. http://stackoverflow.com/a/2256974 (which is in C++ but something similar in node)